wicket-metrics/src/main/java/org/apache/wicket/metrics/WicketMetrics.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.wicket.metrics;
+
+import org.aspectj.lang.ProceedingJoinPoint;
+
+import com.codahale.metrics.JmxReporter;
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.Timer.Context;
+
+/**
+ * Base aspect provides access to the metric registry
+ * 
+ * @author Tobias Soloschenko
+ *
+ */
+public class WicketMetrics
+{
+
+	private static MetricRegistry metricRegistry;
+
+	private static boolean enabled = true;
+
+	private static final String PREFIX = "ApacheWicket/";
+
+	/**
+	 * Gets the metric registry
+	 * 
+	 * @return the metric registry
+	 */
+	public static MetricRegistry getMetricRegistry()
+	{
+		if (metricRegistry == null)
+		{
```

Comment:
This needs synchronization.
Or better - initialize the registry in IInitializer#init()